htlcswitch/link.go

```diff
@@ -2625,6 +2615,18 @@ func (l *channelLink) handleStfu(stfu *lnwire.Stfu) error {
 	return l.quiescer.drive()
 }
 
+// stfuFailf fails the link in the case where the requirements of the quiescence
+// protocol are violated. In all cases we opt to drop the connection as only
+// link state (as opposed to channel state) is affected.
+func (l *channelLink) stfuFailf(format string, args ...interface{}) {
```

Comment:
Do you prefer reviewing smaller commits or smaller numbers of commits? I personally don't mind diff churn, but would rather reason locally about each commit being safe and I find that harder to do the larger the commit is. wdyt?

Reply:
When possible smaller commits, but more pragmatically my personal preference is:
* Only one complete logical idea per commit
* Introduce helpers etc in the earliest commit possible, so that we can reduce the diff (vs introducing some code then later refactoring to a helper in the same PR)

But also lmk if I'm nitpickin too hard on this :')

Reply:
I'll say that the *hardest* changes to incorporate are (some of) the requests to change commit history. In some cases it is easy and I'm happy to do it. In others it is hard, so I'd say if you're flexible wrt me saying no on certain history alterations I think we're fine. I don't think you are nitpicking too hard. Your reviews have a very good 💡/🙄 ratio.

Reply:
Turns out this was easy enough so I did it.